### PR TITLE
CPDRP-770 SIT mentor validation

### PR DESCRIPTION
**Pull request by tonyheadford**
### Context
[Jira](https://dfedigital.atlassian.net/browse/CPDRP-770)
Add changes to participant validation journey for SIT/mentors

### Changes proposed in this pull request
Redirect SIT/mentors to the validation journey if not already completed on sign in.
Add extra step to process for SIT/mentors to ask whether they want to add their mentor information now or not
Add banner to that is shown to SIT/mentors if they have not completed their mentor registration
Changes to end pages for SITs

### Guidance to review
Sign in as a SIT/mentor (`cpd-test+tutor-17@example.com` is a SIT mentor).
You should be shown the "Do you want to add your mentor information" question
Selecting yes, takes you into the participant validation as normal
Selecting no, takes you to the SIT dashboard.
The SIT should see an information banner reminding them to add their mentor information until they complete the validation.

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -63,6 +63,10 @@ def participant?
     early_career_teacher? || mentor?
   end
 
+  def sit_mentor?
+    induction_coordinator? && mentor?
+  end
```
Comment (ethanmills) [inline app/models/user.rb:68]: Do we expect this to be useful often? I'd be tempted to inline it for now

File: spec/forms/participants/participant_validation_form_spec.rb [hunk 2]
```diff
@@ -5,6 +5,31 @@
...
+    context "when no choice has been made" do
+      it "returns false" do
+        expect(form.valid?(:do_you_want_to_add_mentor_information)).to be false
+        expect(form.errors).to include :do_you_want_to_add_mentor_information_choice
+      end
+    end
+
+    context "when an invalid choice has been made" do
+      it "returns false" do
+        form.do_you_know_your_trn_choice = :banana
```
Comment (ethanmills) [inline spec/forms/participants/participant_validation_form_spec.rb:26]: 🍌

**Review by ethanmills — APPROVED** [review]
Happy with this, just a couple of style comments - feel free to ignore

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -6,6 +6,8 @@ def profile_dashboard_path(user)
       admin_schools_path
     elsif user.finance?
       finance_lead_providers_path
+    elsif user.sit_mentor?
```
[the remark this answers is not in the document]
Comment (tonyheadford) [reply app/helpers/application_helper.rb:9]: Think having it helps but happy to change the name, `induction_tutor_and_mentor?` sounds better to me

**Pull request merged**